### PR TITLE
Update limitations of SQLite

### PR DESCRIPTION
**Pull request by agoerler**
We do not document limitations wrt. RIGHT|FULL OUTER JOINs and UNIONs any longer. These SQL constructs are not used by the CAP Java runtime. The limitations could be observed in view definitions and hence rather have to be documented in the CQL area.

### CODE REVIEW

**Review by bugwelle — APPROVED** [review]
LGTM, thanks Adrian!

File: java/cqn-services/persistence-services.md [hunk 1]
```diff
@@ -83,16 +83,14 @@ CAP supports [SQLite](https://www.sqlite.org/index.html) out of the box. When wo
...
-8. Views in SQLite are read-only. However, the CAP Java SDK supports some views to be updatable as described [here](../working-with-cql/query-execution#updatable-views).
-9. Foreign key constraints are supported, but disabled by default. To activate the feature using JDBC URL, append the `foreign_keys=on` parameter to the connection URL, for example, `url=jdbc:sqlite:file:testDb?mode=memory&foreign_keys=on`. For more information, visit the [SQLite Foreign Key Support](https://sqlite.org/foreignkeys.html) in the official documentation.
-10. CAP enables regular expressions on SQLite via a Java implementation. The matching behaviour is an equivalent of the `Matcher.find()` call for the given pattern.
+1. SQLite has only limited support for concurrent database access. You're advised to limit the connection pool to *1* as shown above (parameter `maximum-pool-size: 1`), which effectively serializes all database transactions.
+2. The predicate function `contains` is supported. However, the search for characters in the word or phrase is case-insensitive in SQLite. In the future, we might provide an option to make the case-sensitivity locale dependent.
```
Comment (renejeglinsky) [inline java/cqn-services/persistence-services.md:87]:
```suggestion
2. The predicate function `contains` is supported. However, the search for characters in the word or phrase is case-insensitive in SQLite. In the future, we might provide an option to make the case-sensitivity depending on the locale.
```

File: java/cqn-services/persistence-services.md [hunk 2]
```diff
@@ -83,16 +83,14 @@ CAP supports [SQLite](https://www.sqlite.org/index.html) out of the box. When wo
...
+3. SQLite doesn't support [pessimistic locking](../working-with-cql/query-execution#pessimistic-locking).
+4. Streaming of large object data isn't supported by SQLite. Hence, when reading or writing data of type `cds.LargeString` and `cds.LargeBinary` as a stream the framework temporarily materializes the content. Thus, storing large objects on SQLite can impact the performance.
```
Comment (renejeglinsky) [inline java/cqn-services/persistence-services.md:89]:
```suggestion
4. Streaming of large object data isn't supported by SQLite. Hence, when reading or writing data of type `cds.LargeString` and `cds.LargeBinary` as a stream, the framework temporarily materializes the content. Thus, storing large objects on SQLite can impact the performance.
```

File: java/cqn-services/persistence-services.md [hunk 3]
```diff
@@ -83,16 +83,14 @@ CAP supports [SQLite](https://www.sqlite.org/index.html) out of the box. When wo
...
+5. Sorting of character-based columns is never locale-specific but if any locale is specified in the context of a query then case insensitive sorting is performed.
+6. Views in SQLite are read-only. However, the CAP Java SDK supports some views to be updatable as described [here](../working-with-cql/query-execution#updatable-views).
```
Comment (renejeglinsky) [inline java/cqn-services/persistence-services.md:91]:
```suggestion
6. Views in SQLite are read-only. However, the CAP Java SDK supports some views to be updatable as described in [Updatable Views](../working-with-cql/query-execution#updatable-views).
```

File: java/cqn-services/persistence-services.md (hunk 1 above)
Comment (bugwelle) [reply java/cqn-services/persistence-services.md:87]:
```suggestion
2. The predicate function `contains` is supported. However, the search for characters in the word or phrase is case-insensitive in SQLite. In the future, we might provide an option to make the case-sensitivity depend on the locale.
```

I don't think `depending` fits here, or does it? :)

Comment (renejeglinsky) [reply java/cqn-services/persistence-services.md]: What do we want say? Maybe rephrasing helps:

_In the future, we might provide an option to enable case-sensitivity in search according to the users locale._

Is that correct?

Comment (bugwelle) [reply java/cqn-services/persistence-services.md]: Yes, I think so. :)

Comment (agoerler) [reply java/cqn-services/persistence-services.md]: > In the future, we might provide an option to enable case-sensitivity in search according to the users locale.

I am wondering what the benefit of this information is. I actually think we can just delete this sentence.

File: java/cqn-services/persistence-services.md [hunk 4]
```diff
@@ -83,16 +83,14 @@ CAP supports [SQLite](https://www.sqlite.org/index.html) out of the box. When wo
...
-8. Views in SQLite are read-only. However, the CAP Java SDK supports some views to be updatable as described [here](../working-with-cql/query-execution#updatable-views).
-9. Foreign key constraints are supported, but disabled by default. To activate the feature using JDBC URL, append the `foreign_keys=on` parameter to the connection URL, for example, `url=jdbc:sqlite:file:testDb?mode=memory&foreign_keys=on`. For more information, visit the [SQLite Foreign Key Support](https://sqlite.org/foreignkeys.html) in the official documentation.
-10. CAP enables regular expressions on SQLite via a Java implementation. The matching behaviour is an equivalent of the `Matcher.find()` call for the given pattern.
+1. SQLite has only limited support for concurrent database access. You're advised to limit the connection pool to *1* as shown above (parameter `maximum-pool-size: 1`), which effectively serializes all database transactions.
+2. The predicate function `contains` is supported. However, the search for characters in the word or phrase is case-insensitive in SQLite. In the future, we might provide an option to make the case-sensitivity depend on the locale.
```
Comment (renejeglinsky) [inline java/cqn-services/persistence-services.md:87]:
```suggestion
2. The predicate function `contains` is supported. However, the search for characters in the word or phrase is case-insensitive in SQLite.
```

**Review by renejeglinsky — APPROVED** [review]

**Pull request merged**